examples/example_server_forum.py

```diff
@@ -0,0 +1,155 @@
+        print("")
+        print("GET for", self.client_address)
+        print("Path:", self.path)
+
+        output_txt = ""
+        data_dict = {}
+        try:
+            # params is everything after "?"
+            params = self.path[self.path.index("?")+1:]
+            data_dict = urllib.parse.parse_qs(params)
+        finally:
+            print("Request params:", data_dict)
+
+        try:
+            fd = open(posts_file, "r")
+            # If no user specified, get all comments.
+            if not 'user' in data_dict:
+                output_txt = fd.read()
+            # Fetch comments from specified user
+            else:
+                comment = ""
+                cnt = 0
+                fd = open(posts_file, "r")
+                while True:
+                    comment = fd.readline()
+                    if not comment:
+                        break
+                    cmt_json = read_simple_json(comment, 0)
+                    #print(cmt_json)
+                    if(cmt_json['user'] == data_dict['user'][0]):
+                        cnt += 1
+                        output_txt += comment
+                print("Number of matches:", cnt)
+                if(cnt == 0):
+                    output_txt = "This user has no comments."
+        except:
+            output_txt += ("Error opening file \"" + posts_file + "\"\n")
+            output_txt += "Did you make sure it exists? Maybe post something first.\n"
+            print(output_txt)
+
+        self.send_response(200, message="Ok")
+        self.send_header("Content-Type", "text/plain")
+        self.end_headers()
+        self.wfile.write(output_txt.encode('utf-8'))
+        #self.wfile.write(jsonify("John", "Response").encode('utf-8'))
+        
+
+    # currently just act kinda like an echo server
+    def do_POST(self):
+        print("")
+        print("POST for", self.client_address)
+        print("Request line:", self.requestline)
+        #print(self.headers) # stuff like Host and Content-Type
+        data = self.rfile.read(int(self.headers.get('Content-Length'))).decode('utf-8')
+        print(data)
+        #print("Printed POST data")
+
+        # this handles weird url special character stuff, turns params into dict
+        data_dict = urllib.parse.parse_qs(data)
+        print(data_dict)
+
+        user = data_dict['user'][0]
+        msg = data_dict['msg'][0]
+
+        #write to file
+        fd = open(posts_file, "a")
+        try:
+            fd.write(jsonify(user, msg) + "\n")
+        finally:
+            fd.close()
+
+        self.send_response(200, message="Ok")
+        self.send_header("Content-Type", "text/plain")
+        self.end_headers()
+        self.wfile.write(("Message Posted:\n" + jsonify(user, msg)).encode('utf-8'))
+
+
+# https://stackoverflow.com/questions/19434947/python-respond-to-http-request
+# https://docs.python.org/3/library/http.server.html
+def httpServer(port):
+    print("Starting HTTP Server")
```

Comment:
Would be nice to also print the port it's listening on, just as an extra sanity check